src/main/java/hudson/remoting/Launcher.java

```diff
@@ -127,17 +121,25 @@ public void setTextMode(boolean b) {
         System.out.println("Running in "+mode.name().toLowerCase(Locale.ENGLISH)+" mode");
     }
 
+    /**
+     * @deprecated use {@link #secret}, {@link #name}, {@link #urls}, {@link #webSocket}, {@link #tunnel},
+     * {@link #workDir}, {@link #internalDir}, and/or {@link #failIfWorkDirIsMissing} directly.
+     */
     @Option(name="-jnlpUrl",usage="instead of talking to the controller via stdin/stdout, " +
             "emulate a JNLP client by making a TCP connection to the controller. " +
-            "Connection parameters are obtained by parsing the JNLP file.")
+            "Connection parameters are obtained by parsing the JNLP file.", forbids = {"-direct", "-name", "-tunnel", "-url", "-webSocket"})
+    @Deprecated
     public URL agentJnlpURL = null;
 
-    @Option(name="-jnlpCredentials",metaVar="USER:PASSWORD",usage="HTTP BASIC AUTH header to pass in for making HTTP requests.")
+    @Option(name="-credentials",metaVar="USER:PASSWORD",aliases="-jnlpCredentials",usage="HTTP BASIC AUTH header to pass in for making HTTP requests.")
     public String agentJnlpCredentials = null;
 
     @Option(name="-secret", metaVar="HEX_SECRET", usage="Agent connection secret to use instead of -jnlpCredentials.")
```

Comment:
```suggestion
    @Option(name="-secret", metaVar="HEX_SECRET", usage="Agent connection secret to use instead of -credentials.")
```
since `-jnlpCredentials` is now deprecated; or simply
```suggestion
    @Option(name="-secret", metaVar="HEX_SECRET", usage="Agent connection secret.")
```
since using `-secret` is the normal usage and `-credentials` should perhaps just be deprecated (perhaps there are use cases in exotic reverse proxy configurations, but even then you probably _also_ need `-secret` for the TCP connection).